kmp/reporters.go

```diff
@@ -111,12 +111,24 @@ func (r *ShortDiffReporter) Report(rs cmp.Result) {
 	t := cur.Type()
 	var diff string
 	// Prefix struct values with the types to add clarity in output
-	if !vx.IsValid() || !vy.IsValid() {
-		r.err = fmt.Errorf("Unable to diff %+v and %+v on path %#v", vx, vy, r.path)
+	if !vx.IsValid() && !vy.IsValid() {
+		r.err = fmt.Errorf("Both are invalid on path %#v", r.path)
 	} else if t.Kind() == reflect.Struct {
```

Comment:
It probably makes sense now to reverse the if() conditions to reduce the number of duplicated lines.

i.e.
```
diff = fmt.Sprintf("%#v:\n", r.path)
if vx.IsValid() {
    if t.Kind() == reflect.Struct {
          diff += ...
    } else {
          diff += ...
   }
}

if vy.IsValid() {
    if t.Kind() == reflect.Struct {
          diff += ...
    } else {
          diff += ...
   }
}
```

Reply:
I extract this block into method now, please review again.
```
    if t.Kind() == reflect.Struct {
          diff += ...
    } else {
          diff += ...
   }
```